### ISSUE TITLE
CA2000 Should recognize a pattern for IAsyncDisposable

### ISSUE THREAD

**Issue by RenderMichael** [dotnet/roslyn#69026]
**Brief description:**
There's a pattern with regards to `IAsyncDisposable` when one wants to call `ConfigureAwait(false)` which has been recommended in [official guidance](https://devblogs.microsoft.com/dotnet/configureawait-faq/#can-i-use-configureawait-when-await-using-an-iasyncdisposable) but CA2000 does not respect:
``` csharp
var c = new MyAsyncDisposableClass(); // CA2000: Dispose objects before losing scope
await using (c.ConfigureAwait(false))
{
    ...
}
```

**Languages applicable:**

I am unfamiliar with any language besides C#, maybe this applies to VB or F#.

**Code example that the analyzer should report:**

``` csharp
var c = new MyAsyncDisposableClass();
await using (c.ConfigureAwait(false))
{
    ...
}
```

**Additional information:**

Wars have been fought over language-level support for `ConfigureAwait`. As it stands, this is what we have for this scenario.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).
